ansible/playbooks/roles/upgrade/tasks/kubernetes.yml

```diff
@@ -12,13 +12,22 @@
   import_tasks: kubernetes/get-kubelet-version.yml # sets kubelet_version
   delegate_to: "{{ groups.kubernetes_master[0] }}"
 
+- name: Check for upgrade flag file
+  stat:
+    path: "{{ kubernetes.upgrade_flag_path }}"
+  register: lock_file_status
+
 - name: Upgrade masters then nodes
   vars:
     version: "{{ ver }}"
     cni_version: "{{ cni_ver }}"
   block:
+    - name: Create upgrade flag file
+      file:
+        path: "{{ kubernetes.upgrade_flag_path }}"
+        state: touch
+
     - name: Upgrade masters
-      when: cluster_version is version('v' + version, '<=')
```

Comment:
Here and below we have single conditions, while common one for the whole block may be passed in different cases. Correct me if this is a wrong example/case: upgrade process at this point because of `kubelet_version`, but we don't need to execute this task.

Reply:
> Have you checked creation of files for each incremental upgrade?

Yes, I did, works correctly.

> Here and below we have single conditions, while common one for the whole block may be passed in different cases. Correct me if this is a wrong example/case: upgrade process at this point because of kubelet_version, but we don't need to execute this task

Do you mean that we should separate **Upgrde masters** and **Upgrade nodes** tasks, so if only _kubelet_version_ is lower (_cluster_version_ is already desired) we will execute only **Upgrade nodes** task, and we will not execute **Upgrade masters**?

I was thinking about it and:
- based on fact that kubeadm clusters upgrade has to follow the workflow (primary cp node, additional cp nodes, worker nodes) and
- idempotency character of `kubeadm upgrade` command

I assumed that they can be coupled together.

Reply:
@to-bar what is your opinion here?

Reply:
Cool, I will change it then :)

Reply:
I moved PR to draft since I am not happy yet with solution provided for this part.

Reply:
So finally I just moved state file check for masters and nodes. Inside upgrade of masters, the upgrade of first master and next master will be based on inventory check.